packages/block-editor/src/components/link-control/index.js

```diff
@@ -156,6 +186,19 @@ function LinkControl( {
 		return couldBeURL && ! args.isInitialSuggestions ? results[ 0 ].concat( results[ 1 ] ) : results[ 0 ];
 	};
 
+	/**
+	 * Cancels editing state and marks that focus may need to be restored after
+	 * the next render, if focus was within the wrapper when editing finished.
+	 */
+	function stopEditing() {
+		isEndingEditWithFocus.current = (
+			wrapperNode.current &&
+			wrapperNode.current.contains( document.activeElement )
+		);
```

Comment:
It's not clear to me in which situations `isEndingEditWithFocus` is necessary (compared to refocusing every time we stop editing)

Reply:
> It's not clear to me in which situations `isEndingEditWithFocus` is necessary (compared to refocusing every time we stop editing)

Strictly speaking? It's probably not necessary. In earlier implementations, I had simply set this to `true`.

But, thinking about it more:

- Rather than bind the implementation to assumptions about specific user flows, target the logic closer to the actual problem: If we have focus before a render and expect we lose it after, phrase (and limit) the logic accordingly.
- There's a nice duality in the association between this condition and the one in `useEffect` (essentially its negated form), which mirrors the real-world implications of the same idea of "if we had focus before ➡️ make sure we have focus after".